packages/hdwallet-keepkey/src/osmosis.ts

```diff
@@ -0,0 +1,275 @@
+import type { AminoSignResponse, OfflineAminoSigner, StdSignDoc } from "@cosmjs/amino";
+import type { AccountData } from "@cosmjs/proto-signing";
+import * as Messages from "@keepkey/device-protocol/lib/messages_pb";
+import * as OsmosisMessages from "@keepkey/device-protocol/lib/messages-osmosis_pb";
+import * as core from "@shapeshiftoss/hdwallet-core";
+import * as bs58check from "bs58check";
+import PLazy from "p-lazy";
+
+import { Transport } from "./transport";
+
+const protoTxBuilder = PLazy.from(() => import("@shapeshiftoss/proto-tx-builder"));
+
+export async function osmosisSupportsLegacyAminoSigning(): Promise<boolean> {
+  return true;
+}
+
+export async function osmosisSupportsProtobufSigning(): Promise<boolean> {
+  return false;
+}
```

Comment:
- Where are these used? There doesn't seem to be an equivalent in the cosmos implementation.
- We should probably stick with the `SignModeDirect` and `SignModeLegacyAminoJson` nomenclature Cosmos uses internally to avoid confusion.

Reply:
👍